index.js

```diff
@@ -15,13 +15,30 @@ import RealmProvider from "providers/RealmProvider";
 import React from "react";
 import { AppRegistry } from "react-native";
 import Config from "react-native-config";
+import { setJSExceptionHandler, setNativeExceptionHandler } from "react-native-exception-handler";
 import { GestureHandlerRootView } from "react-native-gesture-handler";
 import { startNetworkLogging } from "react-native-network-logger";
 import { MD3LightTheme as DefaultTheme, Provider as PaperProvider } from "react-native-paper";
 import { SafeAreaProvider } from "react-native-safe-area-context";
 import colors from "styles/tailwindColors";
 
 import { name as appName } from "./app.json";
+import { log } from "./react-native-logs.config";
+
+const rootLog = log.extend( "root" );
+
+const jsErrorHandler = ( e, isFatal ) => {
+  rootLog.error( `JS Error: ${isFatal ? "Fatal:" : ""} ${e.name} ${e.message}` );
```

Comment:
This is great, but when I trigger a simple syntax error, like adding a line that's just `f` in Obs Detail, the log looks like

```
6:17:50 PM | root | ERROR : JS Error:  undefined undefined 
6:17:50 PM | root | ERROR : JS Error: Fatal: ReferenceError Property 'f' doesn't exist 
```

So there are two errors, one has a `name` and `message`, but the other doesn't. If I add `rootLog.info( "e: ", e );`, the error without those properties prints a full stack trace.

So I guess two questions:

1. Why are there two errors?
2. If we can access the stack trace in the second error, can we please log that too? If not, is there some way to consolidate them so we're not logging twice every time we get an exception?

Reply:
I believe the 2nd error is an issue with the `react-native-exception-handler` library rather than our own code, so I added a line to skip error reporting if both the error name and message are undefined.

And added stack trace to the log file.